plugin/httptrace/clienttrace.go

```diff
@@ -76,25 +91,34 @@ func NewClientTrace(ctx context.Context) *httptrace.ClientTrace {
 }
 
 func (ct *clientTracer) start(hook, spanName string, attrs ...core.KeyValue) {
-	_, sp := ct.tr.Start(ct.Context, spanName, trace.WithAttributes(attrs...), trace.WithSpanKind(trace.SpanKindClient))
 	ct.mtx.Lock()
 	defer ct.mtx.Unlock()
-	if ct.root == nil {
-		ct.root = sp
+
+	var ctx context.Context
+	var ok bool
+
+	if ctx, ok = ct.activeHooks[parentHook(hook)]; !ok {
+		ctx = ct.Context
 	}
-	if _, ok := ct.activeHooks[hook]; ok {
-		// end was called before start is handled.
-		sp.End()
-		delete(ct.activeHooks, hook)
+
+	if _, found := ct.activeHooks[hook]; !found {
```

Comment:
You could store the context as `hookCtx` and…

Reply:
Indeed. This is what happens when rearranging code and forgetting to take a step back :)